### ISSUE TITLE
输入的维度可以不一致吗？

### ISSUE THREAD

**Issue by timtian12** [TJCVRS/CRNN_Tensorflow#77]
先非常感谢楼主的分享，然后请教两个问题。
1、代码中我看到需要把输入的图像resize到（30，100）最后输出的是（1*25），然后送入到 lstm中的time steps 最大值是25，那么是否意味着，只能处理最多25个字符的预测？如果我们的输入和输出都是不定长的这种情况下，该怎么办呢？

2、关于构造数据的问题，通过从tfrecord中得到的label的类型时sparse tensor。但是，我查看构造训练集的代码时，发现输入到tfrecord中的label 类型时 int64_list 类型，这里没发现转为sparse tensor的代码？请问这里是怎么实现的呢？

**Comment by TJCVRS** [comment]
@timtian12 模型是可以自动处理变长序列的。第二个问题你可以看看tensorflow的数据序列化方法

**Comment by mdbenito** [comment]
AFAIK, the maximal number of time steps is determined by the number of feature maps after the maxpooling operations of the CNN over the image (in `ShadowNet.__conv_stage()`). Since there are two with stride 2 and kernel size 2 this amounts to dividing the image width by 4, hence length 25 for width 100. We recently added some configuration options to easily change the width and sequence length, but the restriction still applies. @MaybeShewill-CV How can the model manage sequences of arbitrary length?

**Comment by MaybeShewill-CV** [comment]
@mdbenito I think the model can not handle sequence with any length. The length of the sequence must be limited under the sequence max length.

**Comment by mdbenito** [comment]
@MaybeShewill-CV That's what I understood from the paper and the code, but google translate tells me that your previous message says otherwise. I guess Chinese to English translation is not perfect yet! ;)

**Comment by MaybeShewill-CV** [comment]
@mdbenito Yep let's hope the translator will become better under Google's effort. Thanks a lot :)
